src/extensions/nginx-app-protect/monitoring/processor/nap.go

```diff
@@ -336,38 +336,34 @@ func (f *NAPConfig) getMetadata() (*models.Metadata, error) {
 	return NewMetadata(t, f.SupportID)
 }
 
-func (f *NAPConfig) getViolationContext() string {
-	contexts := []string{}
+func (f *NAPConfig) extractViolationsContexts() {
 	if f.ViolationDetailsXML != nil {
 		for i, v := range f.ViolationDetailsXML.RequestViolations.Violations {
 			if v.Context != "" {
```

Comment:
minor: is this check still necessary?

Reply:
Yep, it's still necessary as we don't want to change (extract from violation name) context if it's already been received within XML. Also as olli requested I have reverted back to concatenating and sending `violation_contexts`.